bin/dry-run.js

```diff
@@ -121,25 +122,29 @@ for (const cid of measurementCids) {
 console.log('Fetched %s measurements', round.measurements.length)
 
 console.log('==EVALUATE==')
-const ieContractWithSigner = {
+const ieContract = {
   async getAddress () {
     return contractAddress
-  },
-  async setScores (_roundIndex, participantAddresses, scores) {
-    console.log('==EVALUATION RESULTS==')
-    console.log('participants:', participantAddresses)
-    console.log('scores:', scores)
-    console.log('==END OF RESULTS==')
-    return { hash: '0x234' }
   }
 }
+const signer = ethers.Wallet.createRandom()
+const mockedFetch = async (_, { body }) => {
+  console.log('==EVALUATION RESULTS==')
+  const { participants, scores } = JSON.parse(body)
+  console.log('participants:', participants)
+  console.log('scores:', scores)
+  console.log('==END OF RESULTS==')
+  return new Response()
+}
 
 const started = Date.now()
 const { ignoredErrors } = await evaluate({
   roundIndex,
   round,
   fetchRoundDetails,
-  ieContractWithSigner,
+  ieContract,
+  signer,
+  fetch: mockedFetch,
```

Comment:
IIRC, evaluate calls fetch at other places, e.g. to fetch round details here:

https://github.com/filecoin-station/spark-evaluate/blob/3f93c1289bafaa5a970397e9f9b86e0b6a4176ff/lib/spark-api.js#L18

Let's design & mock this dependency at a higher level.

I propose the following interface:

```ts
function submitEvaluationScores(
  participants: string[],
  values: bigint[]
) => Promise<void>
```

Alternatively:

```ts
function submitEvaluationScores(
  participantScores: Record<string, bigint>
) => Promise<void>
```

(BTW, why are we using an Object and not a Map for the participant/score mapping?)

This way, we can have two implementation:
 - a no-op implementation for the dry run, this means the dry run script does not need any burner wallet
 - a production implementation that uses the `signer` initialised from the seed phrase

In the future, it will be easier to swap the spark-rewards implementation for a different one using IPC or other L2 network, as the code inside `evaluate()` will stay the same.

Reply:
Yeah, I was also thinking about introducing a new function for this. Mocking `fetch` isn't fun. +1

> (BTW, why are we using an Object and not a Map for the participant/score mapping?)

I don't think there's a particular reason for this, and can be refactored.